test/distributed/test_c10d_common.py

```diff
@@ -1488,6 +1490,7 @@ def _test_collectives(self, backend):
             (dist.all_reduce,),
```

Comment:
Sorry, I may have missed this earlier. Would passing `self.rank` to reduce and broadcast cause each rank identifying a different root and hang? They should have the same root.

Reply:
Passing self.rank causes the broadcast operation to be sourced from that rank, but it does not hang waiting for other ranks to ACK the broadcast it sent. The same logic applies to reduce so thats why I believe it is not hanging